fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/NormalizedLuceneSearchProvider.java

```diff
@@ -0,0 +1,163 @@
+ */
+public class NormalizedLuceneSearchProvider extends LuceneSearchProvider implements SearchQuerySearchProvider {
+    private static final Logger LOG = LoggerFactory.getLogger(NormalizedLuceneSearchProvider.class);
+
+    public static final String SEARCH_COLUMN_NAME = "__search";
+    public static final Analyzer DIACRITIC_ANALYZER = new DiacriticNormalizingAnalyzer();
+
+    private SimpleQueryParser queryParser;
+
+    private boolean searchColumnExists;
+    private IndexSearcher lastIndexSearcher;
+
+    /**
+     * Constructor.
+     *
+     * @param luceneIndexPath  Path to the lucene index files
+     * @param maxResults  Maximum number of allowed results in a page
+     * @param searchTimeout  Maximum time in milliseconds that a lucene search can run
+     */
+    public NormalizedLuceneSearchProvider(String luceneIndexPath, int maxResults, int searchTimeout) {
+        super(luceneIndexPath, maxResults, searchTimeout);
+
+        // override analyzer in LuceneSearchProvider
+        Map<String, Analyzer> analyzerMap = new HashMap<>();
+        analyzerMap.put(SEARCH_COLUMN_NAME, DIACRITIC_ANALYZER);
+        analyzer = new PerFieldAnalyzerWrapper(STANDARD_LUCENE_ANALYZER, analyzerMap);
+
+        this.queryParser = new SimpleQueryParser(analyzer, SEARCH_COLUMN_NAME);
+        this.queryParser.setDefaultOperator(BooleanClause.Occur.MUST);
+    }
+
+    /**
+     * Constructor.  The search timeout is initialized to the default (or configured) value.
+     *
+     * @param luceneIndexPath  Path to the lucene index files
+     * @param maxResults  Maximum number of allowed results in a page
+     */
+    public NormalizedLuceneSearchProvider(String luceneIndexPath, int maxResults) {
+        this(luceneIndexPath, maxResults, LUCENE_SEARCH_TIMEOUT_MS);
+    }
+
+    @Override
+    public Pagination<DimensionRow> findSearchRowsPaged(
+            String searchQueryString,
+            PaginationParameters paginationParameters
+    ) {
+       initializeIndexSearcher();
+       if (lastIndexSearcher != luceneIndexSearcher) {
+            synchronized (this) {
+                if (lastIndexSearcher != luceneIndexSearcher) {
+                    lastIndexSearcher = luceneIndexSearcher;
+                    searchColumnExists = validateSearchColumn(SEARCH_COLUMN_NAME);
+                }
+            }
+        }
+        if (!searchColumnExists) {
+            throw new UnsupportedOperationException(
+                    String.format(
+                            "Underlying LuceneIndex for Dimension %s does not support search queries.",
+                            getDimension()
+                    )
+            );
+        }
+
+        return getResultsPage(getSearchQuery(searchQueryString), paginationParameters);
+    }
+
+    @Override
+    public void setDimension(Dimension dimension) {
+        super.setDimension(dimension);
+
+        analyzer = buildNewAnalyzerMapFromDimension(getDimension());
```

Comment:
So I guessa constructor alone wouldn't be sufficient. We'll want a protected `setAnalyzer` regardless.